### DISCUSSION TITLE
Receiving a Canceled Order

### DISCUSSION THREAD

**Discussion by A-Jammoul** [VDA5050/VDA5050#159] [Q&A]
Hello,

We're working with VDA 5050 2.0 for autonomous robots.

At the moment we're implementing advanced handling of orders from MC. MC has the capability to prioritize and reassign orders between AGVs based on priority. The situation we're facing is as follows :
- MC sends an order to the AGV.
- MC cancels the order on the AGV to give the same order to another AGV.
- MC decides that the canceled order should be given to the first AGV again, and sends the same order again to the first AGV.
- Since it's the same order, the orderId is the same. In this case,  according to the diagram for accepting orders on AGVs (Page 18), the message has to be discarded. (Same orderId and same orderUpdateId)
- However, since the AGV already received and acknowledged a cancellation message for that order, we argue that the order should be handled as a new order.

My questions are :
- Does MC have to change the orderId for the order? From an MC point of view, it's still the same order, Which means it makes more sense to have the same orderId.

- We are thinking that in this case, the robot should accept the order again reset the status of the actions in the state and start running the order. Can this logic be added to the VDA5050 approach to handling received orders?

Any help would be appreciated, thank you!

**Comment by collani-bosch** [comment]
I think you come into this situation, because in your system a VDA5050 orderId and an identifier for a task (e.g. a transport order) is the same. I strongly recommend to divide this and not to mix it up. You get into a lot of trouble if you cannot distinguish between these two concepts. There are other situations (especially with human interaction) in which you have to send a new order for continuing your task.
The VDA5050 defines the protocol between a single AGV / AMR and MC; not more. You should not rollout the concepts of the protocol to your MC system, even if the wording (e.g. order) suggest this. I suggest: the MC manage tasks and there are on or more VDA5050 orders necessary to perform this task.

**Reply by heloumarc** [reply]
We are working with the assumption that an order sent to a robot is an instance of a transport order in the MC.
Everytime the MC wants to dispatch an order to a robot it generates a unique orderId (a VDA5050 orderId) this then marks the instance of that transport order. 
From a System Design perspective, this instance needs to be completed, this is where the idea of "reusing" the orderId came from. With this assumption we can say that reassigning an order, which is the same occurrence of that order and not a new one, to a robot that was previously handling it is causing that robot to reject the VDA5050 order.
This is why we think using a new orderId for each time we send the same "instance" of a transport order to a robot is misaligned with the intent of the reassignment feature.
We are happy to discuss it in depth in the next VDA5050 meeting.

**Reply by LukasSchwarz4202** [reply]
In my opinion the best way to integrate this in the existing system would be to add a parameter to cancelOrder which tells the AGV to not only cancel the current order but additionally delete all history => _state/orderId=""_.

I do have concerns though, that this will complicate a lot of things. e.g.:

- analyzing Orders.
- handling the state in the vehicle

**Reply by collani-bosch** [reply]
@LukasSchwarz4202 : It think your proposal is feasible. Maybe it would be less complex if the 'cancelOrder' put the orderId of the aborted order in the result field.
